src/Avalonia.Visuals/VisualExtensions.cs

```diff
@@ -34,48 +31,93 @@ public static Point PointToClient(this IVisual visual, PixelPoint point)
         /// <returns>The point in screen coordinates.</returns>
         public static PixelPoint PointToScreen(this IVisual visual, Point point)
         {
-            var p = GetRootAndPosition(visual);
-            return p.Item1.PointToScreen(point + p.Item2);
+            var p = visual.TranslatePoint(point, visual.VisualRoot);
+            return visual.VisualRoot.PointToScreen(p.Value);
+        }
+
+        /// <summary>
+        /// Returns a transform that transforms the visual's coordinates into the coordinates
+        /// of the specified <paramref name="to"/>.
+        /// </summary>
+        /// <param name="from">The visual whose coordinates are to be transformed.</param>
+        /// <param name="to">The visual to translate the coordinates to.</param>
+        /// <returns>
+        /// A <see cref="Matrix"/> containing the transform or null if the visuals don't share a
+        /// common ancestor.
+        /// </returns>
+        public static Matrix? TransformToVisual(this IVisual from, IVisual to)
+        {
+            var common = from.FindCommonVisualAncestor(to);
```

Comment:
It will return null when we can not find the common ancestor. And the TranslatePoint will return null when `visual` and `relativeTo` can not find the common ancestor and can we check the `visual` and `relativeTo` must can find the common ancestor and throw a argument exception in `TranslatePoint`?

Reply:
Sorry, I can't parse this sentence...